projects/distributions/private/primary/energy/TabulatedFluxDistribution.cxx

```diff
@@ -107,34 +191,51 @@ TabulatedFluxDistribution::TabulatedFluxDistribution(double energyMin, double en
     ComputeIntegral();
     if(has_physical_normalization)
         SetNormalization(integral);
+
+    ComputeCDF();
 }
 
 double TabulatedFluxDistribution::SampleEnergy(std::shared_ptr<LI::utilities::LI_random> rand, std::shared_ptr<LI::detector::EarthModel const> earth_model, std::shared_ptr<LI::crosssections::CrossSectionCollection const> cross_sections, LI::dataclasses::InteractionRecord const & record) const {
-    // Metropolis-Hastings algorithm to sample from PDF.
-    // Pass in a function pointer for the PDF
-
-    double energy, density, test_energy, test_density, odds;
-    bool accept;
-
-    // sample an initial point uniformly
-    energy = rand->Uniform(energyMin, energyMax);
-    density = pdf(energy);
-
-    // Metropolis Hastings loop
-    for (size_t j = 0; j <= burnin; ++j) {
-        test_energy = rand->Uniform(energyMin, energyMax);
-        test_density = pdf(test_energy);
-        odds = test_density / density;
-        accept = (odds > 1.) or (rand->Uniform(0,1) < odds);
-        if(accept) {
-            energy = test_energy;
-            density = test_density;
-        }
-    }
+    // inverse CDF algorithm to sample from PDF.
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<> dis(0, 1);
+    // Generate a random number from a uniform distribution [0, 1]
+    double randomValue = dis(gen);
```

Comment:
This function should use the supplied `std::shared_ptr<LI::utilities::LI_random>` object instead of creating a new random number generator on each call.